test/unit/test_randomai.gd

```diff
@@ -338,28 +338,33 @@ func run_ai_game():
 
 		var turn_events = []
 		turn_events += handle_decisions(game_logic) #Handles overdrives
-		var turn_action = current_ai.take_turn(game_logic, current_player.my_id)
-		if turn_action is AIPlayer.PrepareAction:
-			turn_events += handle_prepare(game_logic, current_player)
-		elif turn_action is AIPlayer.MoveAction:
-			turn_events += handle_move(game_logic, current_player, turn_action)
-		elif turn_action is AIPlayer.ChangeCardsAction:
-			turn_events += handle_change_cards(game_logic, current_player, turn_action)
-		elif turn_action is AIPlayer.ExceedAction:
-			turn_events += handle_exceed(game_logic, other_ai, current_player, turn_action)
-		elif turn_action is AIPlayer.ReshuffleAction:
-			turn_events += handle_reshuffle(game_logic, current_player)
-		elif turn_action is AIPlayer.BoostAction:
-			turn_events += handle_boost(game_logic, current_ai, other_ai, current_player, turn_action)
-		elif turn_action is AIPlayer.StrikeAction:
-			turn_events += handle_strike(game_logic, current_ai, other_ai, turn_action)
-		elif turn_action is AIPlayer.CharacterActionAction:
-			turn_events += handle_character_action(game_logic, current_ai, other_ai, turn_action)
-		else:
-			fail_test("Unknown turn action: %s" % turn_action)
-			assert(false, "Unknown turn action: %s" % turn_action)
+
+		if game_logic.game_state != Enums.GameState.GameState_WaitForStrike:
+			var turn_action = current_ai.take_turn(game_logic, current_player.my_id)
+			if turn_action is AIPlayer.PrepareAction:
+				turn_events += handle_prepare(game_logic, current_player)
+			elif turn_action is AIPlayer.MoveAction:
+				turn_events += handle_move(game_logic, current_player, turn_action)
+			elif turn_action is AIPlayer.ChangeCardsAction:
+				turn_events += handle_change_cards(game_logic, current_player, turn_action)
+			elif turn_action is AIPlayer.ExceedAction:
+				turn_events += handle_exceed(game_logic, other_ai, current_player, turn_action)
+			elif turn_action is AIPlayer.ReshuffleAction:
+				turn_events += handle_reshuffle(game_logic, current_player)
+			elif turn_action is AIPlayer.BoostAction:
+				turn_events += handle_boost(game_logic, current_ai, other_ai, current_player, turn_action)
+			elif turn_action is AIPlayer.StrikeAction:
+				turn_events += handle_strike(game_logic, current_ai, other_ai, turn_action)
+			elif turn_action is AIPlayer.CharacterActionAction:
+				turn_events += handle_character_action(game_logic, current_ai, other_ai, turn_action)
+			else:
+				fail_test("Unknown turn action: %s" % turn_action)
+				assert(false, "Unknown turn action: %s" % turn_action)
 
 		turn_events += handle_decisions(game_logic)
+		if game_logic._get_player(game_logic.active_turn_player) != current_player:
```

Comment:
I'm wary of changes in this that I don't fully understand all the implications of.
How many characters have you run this through?
I think this is okay though, as presumably all the below stuff wouldn't make sense if the player turn changed.